packages/shopify-app-session-storage-prisma/src/prisma.ts

```diff
@@ -38,11 +39,24 @@ export class PrismaSessionStorage<T extends PrismaClient>
 
     const data = this.sessionToRow(session);
 
-    await this.getSessionTable().upsert({
-      where: {id: session.id},
-      update: data,
-      create: data,
-    });
+    try {
```

Comment:
I wonder if instead of catching the error, it might make more sense to run upsert / delete operations inside [transactions](https://www.prisma.io/docs/orm/prisma-client/queries/transactions#the-transaction-api)?

Reply:
Well, assuming the DBMS supports locking writes to the table, transactions should do the trick. However you're right in that we should probably not assume that will be the case for every system supported by prisma, right?